Tomas Heran <tomas.heran@oracle.com>
Fix to configure to not use -fno-strict-aliasing and -fvisibility=hidden with
Solaris Studio cc as it doesn't like it for dynamic stuff.
Reported upstream as: https://gnunet.org/bugs/view.php?id=3584

diff -r 4a6b889815b1 -r dcb95cb7b865 configure.ac
--- a/configure.ac	Thu Sep 04 20:38:52 2014 +0200
+++ b/configure.ac	Thu Dec 18 21:16:20 2014 +0100
@@ -248,7 +248,7 @@
 
 # set GCC options
 # use '-fno-strict-aliasing', but only if the compiler can take it
-AX_APPEND_COMPILE_FLAGS([[-fno-strict-aliasing]])
+# AX_APPEND_COMPILE_FLAGS([[-fno-strict-aliasing]])
 
 AC_C_BIGENDIAN
 
@@ -435,23 +435,23 @@
 # TCP_CORK and TCP_NOPUSH
 AC_CHECK_DECLS([TCP_CORK, TCP_NOPUSH], [], [], [[#include <netinet/tcp.h>]])
 
-HIDDEN_VISIBILITY_CFLAGS=""
-case "$host" in
-  *-*-mingw*)
-    dnl on mingw32 we do -fvisibility=hidden and __declspec(dllexport)
-    AC_DEFINE([_MHD_EXTERN], [__attribute__((visibility("default"))) __declspec(dllexport) extern],
-              [defines how to decorate public symbols while building])
-    HIDDEN_VISIBILITY_CFLAGS="-fvisibility=hidden"
-    ;;
-  *)
-    dnl on other compilers, check if we can do -fvisibility=hidden
-    AX_CHECK_COMPILE_FLAG([-fvisibility=hidden],
-                           [AC_DEFINE([_MHD_EXTERN], [__attribute__((visibility("default"))) extern],
-                                       [defines how to decorate public symbols while building])
-                            HIDDEN_VISIBILITY_CFLAGS="-fvisibility=hidden"])
-    ;;
-esac
-AC_SUBST(HIDDEN_VISIBILITY_CFLAGS)
+# HIDDEN_VISIBILITY_CFLAGS=""
+# case "$host" in
+#   *-*-mingw*)
+#     dnl on mingw32 we do -fvisibility=hidden and __declspec(dllexport)
+#     AC_DEFINE([_MHD_EXTERN], [__attribute__((visibility("default"))) __declspec(dllexport) extern],
+#               [defines how to decorate public symbols while building])
+#     HIDDEN_VISIBILITY_CFLAGS="-fvisibility=hidden"
+#     ;;
+#   *)
+#     dnl on other compilers, check if we can do -fvisibility=hidden
+#     AX_CHECK_COMPILE_FLAG([-fvisibility=hidden],
+#                            [AC_DEFINE([_MHD_EXTERN], [__attribute__((visibility("default"))) extern],
+#                                        [defines how to decorate public symbols while building])
+#                             HIDDEN_VISIBILITY_CFLAGS="-fvisibility=hidden"])
+#     ;;
+# esac
+# AC_SUBST(HIDDEN_VISIBILITY_CFLAGS)
 
 # libcurl (required for testing)
 AC_ARG_ENABLE([curl],
